### PR TITLE
feat(): clinical details db acsses

### PR DESCRIPTION
**Pull request by Sigwoman**

### CODE REVIEW

File: server/src/DBService/ClinicalDetails/Mutation.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { gql } from "postgraphile";
+
+export const CREATE_ADDRESS = gql`
+mutation MyMutation ($input: InsertAndGetAddressIdInput!) {
```
Comment (RoeiRom) [inline server/src/DBService/ClinicalDetails/Mutation.ts:4]: change mutation name

File: server/src/DBService/ClinicalDetails/Mutation.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    __typename
```
Comment (RoeiRom) [inline server/src/DBService/ClinicalDetails/Mutation.ts:5]: remove typename

File: server/src/DBService/ClinicalDetails/Mutation.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    insertAndGetAddressId(input: $input) {
+      integer
+    }
+  }
+   
+`;
+
+export const CREATE_INVESTIGATION = gql`
+mutation AddInvestigation ($investigation: InvestigationInput!) {
+    __typename
```
Comment (RoeiRom) [inline server/src/DBService/ClinicalDetails/Mutation.ts:15]: remove typename

File: server/src/DBService/ClinicalDetails/Mutation.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    createInvestigation(input: {investigation: $investigation}) {
+      clientMutationId
+    }
+  }
+`;
+
+export const ADD_BACKGROUND_DESEASES = gql`
+mutation MyMutation ($backgroundDeseases: [String!], $investigatedPatientId: Int!) {
```
Comment (RoeiRom) [inline server/src/DBService/ClinicalDetails/Mutation.ts:23]: change mutation name

File: server/src/DBService/ClinicalDetails/Mutation.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    __typename
+    insertBackgroundDeseases(input: {backgroundDeseases: $backgroundDeseases, investigatedPatientId: $investigatedPatientId}) {
+      integer
+    }
+  }
+`;
+
+export const ADD_SYMPTOMS = gql`
+mutation MyMutation ($investigationIdValue: Int!, $symptomNames: [String!]) {
```
Comment (RoeiRom) [inline server/src/DBService/ClinicalDetails/Mutation.ts:32]: change mutation name

File: server/src/DBService/ClinicalDetails/Mutation.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    __typename
+    insertSymptoms(input: {investigationIdValue: $investigationIdValue, symptomNames: $symptomNames}) {
```
Comment (RoeiRom) [inline server/src/DBService/ClinicalDetails/Mutation.ts:34]: where is the function script

File: server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts [hunk 7]
```diff
@@ -1,9 +1,58 @@
...
+            investigatedPatientId: requestInvestigation.investigatedPatientId,
+            investigationStatus: requestInvestigation.investigationStatus,
+            isInIsolation: requestInvestigation.isInIsolation,
+            isIsolationProblem: requestInvestigation.isIsolationProblem,
+            isIsolationProblemMoreInfo: requestInvestigation.isIsolationProblemMoreInfo,
+            isolationStartTime: requestInvestigation.isolationStartTime,
+            lastUpdateTime: requestInvestigation.lastUpdateTime,
+            startTime: requestInvestigation.startTime,
+            symptomsStartTime: requestInvestigation.symptomsStartTime
+        }}).then((result: any) => {
```
Comment (RoeiRom) [inline server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts:42]: if you are not using the result than dont request it in the function, use only ()

File: server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts [hunk 8]
```diff
@@ -1,9 +1,58 @@
...
+
+clinicalDetailsRoute.post('/saveClinicalDetails', (request: Request, response: Response) => {
+    
+    const requestAddress = request.body.input;
+
+    const requestInvestigation = request.body.investigation;
+    
+    graphqlRequest(CREATE_ADDRESS, { input: {
+        ...requestAddress
+    }}).then((result: any) => {
```
Comment (RoeiRom) [inline server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts:20]: add type

**Review by RoeiRom — CHANGES_REQUESTED** [review]

File: server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts (hunk 8 above)
Comment (Sigwoman) [reply server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts:20]: it has no type. each time its diffrent

File: server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts (hunk 7 above)
Comment (Sigwoman) [reply server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts:42]: ok

File: server/src/DBService/ClinicalDetails/Mutation.ts (hunk 1 above)
Comment (Sigwoman) [reply server/src/DBService/ClinicalDetails/Mutation.ts:4]: tzodeck

**Review by RoeiRom — CHANGES_REQUESTED** [review]
add function script!

File: server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts (hunk 8 above)
Comment (RoeiRom) [reply server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts:20]: how is this possible, you are always returning the same structure

File: server/src/DBService/ClinicalDetails/DbScripts.sql [hunk 9]
```diff
@@ -0,0 +1,111 @@
+-- FUNCTION: public.insert_and_get_address_id(character varying, character varying, character varying, integer, integer, character varying, integer)
+
+-- DROP FUNCTION public.insert_and_get_address_id(character varying, character varying, character varying, integer, integer, character varying, integer);
+
+CREATE OR REPLACE FUNCTION public.insert_and_get_address_id(
+	city_value character varying,
+	neighbourhood_value character varying,
+	street_value character varying,
+	house_num_valuea integer,
```
Comment (RoeiRom) [inline server/src/DBService/ClinicalDetails/DbScripts.sql:9]: value*

File: server/src/DBService/ClinicalDetails/DbScripts.sql [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+AS $BODY$
+DECLARE background_deseas char varying;
+BEGIN
+	FOREACH background_deseas IN ARRAY background_deseases
+	LOOP
+		INSERT INTO investigated_patient_background_diseases
+		(investigated_patient_id, background_deseas_name)
+		VALUES (investigated_patient_id, background_deseas);
+	END LOOP;
+	RETURN cardinality(background_deseases);
```
Comment (RoeiRom) [inline server/src/DBService/ClinicalDetails/DbScripts.sql:78]: why returning the length of the array? you can return void like regular function

**Review by RoeiRom — CHANGES_REQUESTED** [review]

File: server/src/DBService/ClinicalDetails/DbScripts.sql (hunk 10 above)
Comment (Sigwoman) [reply server/src/DBService/ClinicalDetails/DbScripts.sql:78]: how?

Comment (RoeiRom) [reply server/src/DBService/ClinicalDetails/DbScripts.sql:78]: you say the function returns void and you dont write return in the function

File: server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts (hunk 8 above)
Comment (Sigwoman) [reply server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts:20]: you are right, but i dont think we need to create a type for everything. Just like we dont usually create an interface for every data we get from useQuery (we only create interfaces for the actual data). Ata mevin ma ani omer? 
I mean i can create an interface but is it necessary?

Comment (RoeiRom) [reply server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts:20]: i think you can do it (I did it in the axios request)

Comment (yaniryos) [reply server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts:20]: i will check

File: server/src/DBService/ClinicalDetails/DbScripts.sql [hunk 11]
```diff
@@ -75,7 +75,7 @@ BEGIN
 		(investigated_patient_id, background_deseas_name)
 		VALUES (investigated_patient_id, background_deseas);
 	END LOOP;
-	RETURN cardinality(background_deseases);
+	RETURN;
```
Comment (RoeiRom) [inline server/src/DBService/ClinicalDetails/DbScripts.sql:78]: you can remove the return

File: server/src/DBService/ClinicalDetails/DbScripts.sql [hunk 12]
```diff
@@ -103,7 +103,7 @@ BEGIN
 		(investigation_id, symptom_name)
 		VALUES (investigation_id_value, symptom_name);
 	END LOOP;
-	RETURN cardinality(symptom_names);
+	RETURN;
```
Comment (RoeiRom) [inline server/src/DBService/ClinicalDetails/DbScripts.sql:106]: same

**Review by RoeiRom — CHANGES_REQUESTED** [review]

File: server/src/DBService/ClinicalDetails/DbScripts.sql (hunk 11 above)
Comment (yaniryos) [reply server/src/DBService/ClinicalDetails/DbScripts.sql:78]: i cant . i tried didnt work

File: server/src/DBService/ClinicalDetails/DbScripts.sql (hunk 12 above)
Comment (yaniryos) [reply server/src/DBService/ClinicalDetails/DbScripts.sql:106]: i cant . i tried didnt work

**Review by roymaniv — APPROVED** [review]

File: server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts [hunk 13]
```diff
@@ -1,9 +1,59 @@
...
+clinicalDetailsRoute.post('/saveClinicalDetails', (request: Request, response: Response) => {
+    
+    const requestAddress = request.body.input;
+
+    const requestInvestigation = request.body.investigation;
+    
+    graphqlRequest(CREATE_ADDRESS, { input: {
+        ...requestAddress
+    }}).then((result: CreateAddressRespone) => {
+        graphqlRequest(CREATE_INVESTIGATION, { investigation: {
```
Comment (MayaSasson) [inline server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts:22]: cant you spread the requestInvestigation object and than add the isolationAddress to the new object

File: server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts [hunk 14]
```diff
@@ -1,9 +1,59 @@
...
+            startTime: requestInvestigation.startTime,
+            symptomsStartTime: requestInvestigation.symptomsStartTime
+        }}).then(() => {
+            graphqlRequest(ADD_BACKGROUND_DESEASES, {
+                backgroundDeseases: requestInvestigation.backgroundIllnesses,
+                investigatedPatientId: requestInvestigation.investigatedPatientId
+            }).then(() => {
+                graphqlRequest(ADD_SYMPTOMS, {
+                    investigationIdValue: requestInvestigation.epidemioligyNumber,
+                    symptomNames: requestInvestigation
```
Comment (MayaSasson) [inline server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts:50]: requestInvestigation - is that really the symptomNames

**Review by MayaSasson — DISMISSED** [review]

**Review by daniel7764 — APPROVED** [review]

**Review by MayaSasson — APPROVED** [review]

**Review by MayaSasson — APPROVED** [review]

**Review by RoeiRom — APPROVED** [review]

**Pull request merged**